pinot-segment-local/src/test/java/org/apache/pinot/segment/local/customobject/AvgPrecisionPairTest.java

```diff
@@ -0,0 +1,206 @@
+
+    byte[] bytes = original.toBytes();
+    AvgPrecisionPair deserialized = AvgPrecisionPair.fromBytes(bytes);
+
+    assertEquals(deserialized.getSum(), negativeSum);
+    assertEquals(deserialized.getCount(), count);
+  }
+
+  @Test
+  public void testCompareTo() {
+    AvgPrecisionPair pair1 = new AvgPrecisionPair(new BigDecimal("100"), 10L);
+    AvgPrecisionPair pair2 = new AvgPrecisionPair(new BigDecimal("200"), 10L);
+    AvgPrecisionPair pair3 = new AvgPrecisionPair(new BigDecimal("100"), 10L);
+
+    assertTrue(pair1.compareTo(pair2) < 0);
+    assertTrue(pair2.compareTo(pair1) > 0);
+    assertEquals(pair1.compareTo(pair3), 0);
+  }
+
+  @Test
+  public void testCompareToWithDifferentCounts() {
+    // Average of pair1: 100/10 = 10
+    AvgPrecisionPair pair1 = new AvgPrecisionPair(new BigDecimal("100"), 10L);
+    // Average of pair2: 200/10 = 20
+    AvgPrecisionPair pair2 = new AvgPrecisionPair(new BigDecimal("200"), 10L);
+    // Average of pair3: 150/10 = 15
+    AvgPrecisionPair pair3 = new AvgPrecisionPair(new BigDecimal("150"), 10L);
+
+    assertTrue(pair1.compareTo(pair2) < 0);
+    assertTrue(pair2.compareTo(pair3) > 0);
+    assertTrue(pair1.compareTo(pair3) < 0);
+  }
+
+  @Test
+  public void testCompareToWithZeroCount() {
+    AvgPrecisionPair pair1 = new AvgPrecisionPair(new BigDecimal("100"), 0L);
+    AvgPrecisionPair pair2 = new AvgPrecisionPair(new BigDecimal("200"), 0L);
+
+    // Both have zero count, averages are undefined, so they are equal
+    assertEquals(pair1.compareTo(pair2), 0);
+  }
+
+  @Test
+  public void testRoundTripSerializationWithVeryLargeNumbers() {
+    BigDecimal veryLargeSum = new BigDecimal("123456789012345678901234567890.123456789012345678901234567890");
+    long veryLargeCount = Long.MAX_VALUE;
+    AvgPrecisionPair original = new AvgPrecisionPair(veryLargeSum, veryLargeCount);
+
+    byte[] bytes = original.toBytes();
+    AvgPrecisionPair deserialized = AvgPrecisionPair.fromBytes(bytes);
+
+    assertEquals(deserialized.getSum(), veryLargeSum);
+    assertEquals(deserialized.getCount(), veryLargeCount);
+  }
+
+  @Test
+  public void testMultipleApplyCalls() {
+    AvgPrecisionPair pair = new AvgPrecisionPair(BigDecimal.ZERO, 0L);
+
+    pair.apply(new BigDecimal("10"), 1L);
+    pair.apply(new BigDecimal("20"), 2L);
+    pair.apply(new BigDecimal("30"), 3L);
+
+    assertEquals(pair.getSum(), new BigDecimal("60"));
+    assertEquals(pair.getCount(), 6L);
+  }
+
+  @Test
+  public void testPrecisionPreservation() {
+    // Test that BigDecimal precision is preserved through serialization
+    BigDecimal preciseSum = new BigDecimal("0.123456789012345678901234567890");
+    AvgPrecisionPair original = new AvgPrecisionPair(preciseSum, 1L);
+
+    byte[] bytes = original.toBytes();
+    AvgPrecisionPair deserialized = AvgPrecisionPair.fromBytes(bytes);
+
+    assertEquals(deserialized.getSum(), preciseSum);
+    assertEquals(deserialized.getSum().scale(), preciseSum.scale());
```

Comment:
The test says it's checking precision preservation but we're actually only comparing the scale and not the precision?